doc/use_cases.rst

```diff
@@ -0,0 +1,113 @@
+..
+  Copyright (C) 2015  Red Hat, Inc.
+
+  This copyrighted material is made available to anyone wishing to use,
+  modify, copy, or redistribute it subject to the terms and conditions of
+  the GNU General Public License v.2, or (at your option) any later version.
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY expressed or implied, including the implied warranties of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+  Public License for more details.  You should have received a copy of the
+  GNU General Public License along with this program; if not, write to the
+  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+  02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+  source code or documentation are not subject to the GNU General Public
+  License and may only be used or replicated with the express permission of
+  Red Hat, Inc.
+
+###############
+ DNF Use Cases
+###############
+
+.. contents::
+
+==============
+ Introduction
+==============
+
+Every feature present in DNF should be based on a reasonable use case. All the
+supported use cases are supposed to be enumerated in this document.
+
+In case you use DNF to achieve a goal which is not documented here, either you
+found an error in the documentation or you misuse DNF. In either case we would
+appreciate if you share the case with us so we can help you to use DNF in the
+correct way or add the case to the list. You can only benefit from such a
+report because then you can be sure that the behavior that you expect will not
+change without prior notice in the :doc:`release_notes` and that the behavior
+will be covered by our test suite.
+
+.. IMPORTANT::
+
+  Please consult every usage of DNF with our reference documentation to be sure
+  what are you doing. The examples mentioned here are supposed to be as simple
+  as possible and may ignore some minor corner cases.
+
+.. WARNING::
+
+  The list is not complete yet - the use cases are being added incrementally
+  these days.
+
+=====================
+ General assumptions
+=====================
+
+DNF must always preserve the integrity of the related RPMDB.
```

Comment:
I would not mention DNF here. Then users could be afraid to run scripts being afraid of DNF breaking RPMDB. What about `RPMDB should not be corrupted.`